.github/workflows/release-console.yml

```diff
@@ -0,0 +1,88 @@
+name: Release Console
+
+on:
+  workflow_dispatch:
+    inputs:
+      releaseVersion:
+        description: 'Release version (e.g., v1.2.3)'
+        required: true
+      commitHash:
+        description: 'Commit hash to tag'
+        required: true
+      environment:
+        description: 'Deployment environment (main or testnet)'
+        required: true
+        default: 'main'
+
+env:
+  E2E_TEST_WORKFLOW: ./.github/workflows/ci-trading-python-tests.yml
+
+jobs:
+  setup_release_branch:
+    runs-on: ubuntu-latest
+    outputs:
+      branch_name: ${{ steps.prepare_branch.outputs.branch_name }}
+    steps:
+      - name: Checkout code at commit hash
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.commitHash }}
+
+      - name: Create Tag
+        run: git tag ${{ github.event.inputs.releaseVersion }} ${{ github.event.inputs.commitHash }}
+
+      - name: Push Tag
+        run: git push origin ${{ github.event.inputs.releaseVersion }}
+
+      - name: Prepare release branch
+        id: prepare_branch
+        run: |
+          BRANCH_NAME="release/${{ github.event.inputs.environment }}/${{ github.event.inputs.releaseVersion }}"
+          echo "::set-output name=branch_name::$BRANCH_NAME"
+          git fetch --all
+          git checkout -b $BRANCH_NAME origin/${{ github.event.inputs.environment }}
+          git reset --hard ${{ github.event.inputs.commitHash }}
+          git push --force --set-upstream origin $BRANCH_NAME
+        if: ${{ github.event.inputs.environment == 'testnet' }}
+
+  run_e2e_tests:
+    needs: setup_release_branch
+    uses: ${{ env.E2E_TEST_WORKFLOW }}
+    with:
+      github-sha: ${{ needs.setup_release_branch.outputs.branch_name }}
```

Comment:
Aren't we missing some steps ?

Reply:
I don't believe so. Testing on a forked repo this works.